From 01dc0c50c38aded0f6b9b1b978960c1d35652971 Mon Sep 17 00:00:00 2001
From: qianmenx <qianx.meng@intel.com>
Date: Thu, 25 Jul 2019 10:40:28 +0800
Subject: [PATCH] media: intel-ipu4: resolve compile error with allyesconfig

Resolve compile error with kernel upgrade, kernel version
5.3 rc1

Add header file and change one interface function of v4l2
- use vidioc_enum_fmt_vid_cap replace
  vidioc_enum_fmt_vid_cap_mplane
- add header file to dw9714.c as3638.h ipu-isys-isa-fw.h
  ipu-psys.h

Change-Id: I8d13700a703144a0183709342ac29edc9997f55b
Tracked-On: PKT-2690
Tracked-On: #H1507330328
Signed-off-by: qianmenx <qianx.meng@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
Signed-off-by: Yew, Chang Ching <chang.ching.yew@intel.com>
---
 drivers/media/pci/intel/ipu-isys-video.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/media/pci/intel/ipu-isys-video.c b/drivers/media/pci/intel/ipu-isys-video.c
index 942ad17..6684b01 100644
--- a/drivers/media/pci/intel/ipu-isys-video.c
+++ b/drivers/media/pci/intel/ipu-isys-video.c
@@ -1779,7 +1779,11 @@ static long ipu_isys_compat_ioctl(struct file *file, unsigned int cmd,
 
 static const struct v4l2_ioctl_ops ioctl_ops_mplane = {
 	.vidioc_querycap = ipu_isys_vidioc_querycap,
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 3, 0)
+	.vidioc_enum_fmt_vid_cap = ipu_isys_vidioc_enum_fmt,
+#else
 	.vidioc_enum_fmt_vid_cap_mplane = ipu_isys_vidioc_enum_fmt,
+#endif
 	.vidioc_g_fmt_vid_cap_mplane = vidioc_g_fmt_vid_cap_mplane,
 	.vidioc_s_fmt_vid_cap_mplane = vidioc_s_fmt_vid_cap_mplane,
 	.vidioc_try_fmt_vid_cap_mplane = vidioc_try_fmt_vid_cap_mplane,
-- 
1.9.1

